src/inmanta/agent/cache.py

```diff
@@ -37,28 +37,35 @@ class CacheItem:
     def __init__(
         self,
         key: str,
-        timeout: float,
         value: Any,
         call_on_delete: Optional[Callable[[Any], None]],
-        lingering: bool = True,
+        evict_after_last_access: float,
+        evict_after_creation: float,
     ) -> None:
         """
         :param key: The full key identifying this item in the cache.
-        :param timeout: Hard timeout for non-lingering cache item.
         :param value: The value being cached associated to the key.
         :param call_on_delete: Optional finalizer to call when the cache item is deleted. This is
             a callable expecting the cached value as an argument.
-        :param lingering: When True, this cache item will linger in the cache for 60s after its last use.
-            When False, this cache item will be evicted from the cache <timeout> seconds after
+        :param evict_after_creation: This cache item will be considered stale this number of seconds after
```

Comment:
Minor comment: It's a bit weird that the order of parameters is not following the order in the method definition. This comment applies at different places